ipv8/configuration_pydantic.py

```diff
@@ -0,0 +1,357 @@
+    peers: int = 20
+    init: dict = Field(default_factory=dict)
+
+    @validator("peers", always=True)
+    @classmethod
+    def validate_peer_count(cls, v):
+        assert v >= 0 or v == -1, f"A walker's peer count must be >= 0 or set to -1 (infinite), got: {v}!"
+        return v
+
+
+class RandomWalk(Walker):
+    def __init__(self, **kwargs):
+        kwargs["strategy"] = "RandomWalk"
+        if "init" not in kwargs:
+            kwargs["init"] = {"timeout": 3.0}
+        super().__init__(**kwargs)
+
+
+class EdgeWalk(Walker):
+    def __init__(self, **kwargs):
+        kwargs["strategy"] = "EdgeWalk"
+        super().__init__(**kwargs)
+
+
+class RandomChurn(Walker):
+    def __init__(self, **kwargs):
+        kwargs["strategy"] = "RandomChurn"
+        if "peers" not in kwargs:
+            kwargs["peers"] = -1
+        if "init" not in kwargs:
+            kwargs["init"] = {'sample_size': 8, 'ping_interval': 10.0, 'inactive_time': 27.5, 'drop_time': 57.5}
+        super().__init__(**kwargs)
+
+
+class PeriodicSimilarity(Walker):
+    def __init__(self, **kwargs):
+        kwargs["strategy"] = "PeriodicSimilarity"
+        if "peers" not in kwargs:
+            kwargs["peers"] = -1
+        super().__init__(**kwargs)
+
+
+class PingChurn(Walker):
+    def __init__(self, **kwargs):
+        kwargs["strategy"] = "PingChurn"
+        if "peers" not in kwargs:
+            kwargs["peers"] = -1
+        super().__init__(**kwargs)
+
+
+class Overlay(BaseConfigurationClass):
+    cls: str = Field(alias='class')
+    key: str = default["keys"][0]["alias"]
+    walkers: List[Walker] = [RandomWalk()]
+    bootstrappers: List[Bootstrapper] = [DispersyBootstrapper(init=DispersyBootstrapperInit())]
+    initialize: dict = Field(default_factory=dict)
+    on_start: List[tuple] = []
+
+
+class IPv8Configuration(BaseConfigurationClass):
+    """
+    Main pydantic IPv8 configuration model, can be fed directly into the ``IPv8`` class constructor.
+    """
+    interfaces: List[Interface] = [Interface()]
+    key_aliases: List[Key] = Field([FileKey()], alias="keys")
+    logger: Logger = Logger()
+    working_directory: str = default["working_directory"]
+    walker_interval: float = default["walker_interval"]
+    overlays: List[Overlay] = [
+        Overlay(cls="DiscoveryCommunity", walkers=[RandomWalk(), RandomChurn(), PeriodicSimilarity()]),
+        Overlay(cls="HiddenTunnelCommunity", initialize=default["overlays"][1]["initialize"],
+                on_start=default["overlays"][1]["on_start"]),
+        Overlay(cls="DHTDiscoveryCommunity", walkers=[RandomWalk(), PingChurn()])
+    ]
+
+    @validator("interfaces", always=True)
+    @classmethod
+    def validate_interfaces(cls, v):
+        it_names = [spec.interface for spec in v]
+        assert len(v) == len(set(it_names)), f"Duplicate interface names specified: {', '.join(it_names)}!"
+        return v
+
+
+def format_schema_recursive(*base_models: Type[BaseModel]) -> dict:
```

Comment:
Thank you for this method. It is quite helpful for the generation of documentation.